Fix tests failures in ast_test.py

These are the failures in ast_test.py
```
FAILED ../../../../integration_tests/src/main/python/ast_test.py::test_abs
FAILED ../../../../integration_tests/src/main/python/ast_test.py::test_addition
FAILED ../../../../integration_tests/src/main/python/ast_test.py::test_multi_tier_ast
FAILED ../../../../integration_tests/src/main/python/ast_test.py::test_multiplication
FAILED ../../../../integration_tests/src/main/python/ast_test.py::test_subtraction
FAILED ../../../../integration_tests/src/main/python/ast_test.py::test_unary_minus
```

--- a/integration_tests/src/main/python/ast_test.py
+++ b/integration_tests/src/main/python/ast_test.py
@@ -1,4 +1,4 @@
-# Copyright (c) 2021-2023, NVIDIA CORPORATION.
+# Copyright (c) 2021-2024, NVIDIA CORPORATION.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -16,7 +16,7 @@ import pytest
 
 from asserts import assert_cpu_and_gpu_are_equal_collect_with_capture
 from data_gen import *
-from marks import approximate_float, datagen_overrides, ignore_order
+from marks import approximate_float, datagen_overrides, ignore_order, disable_ansi_mode
 from spark_session import with_cpu_session, is_before_spark_330
 import pyspark.sql.functions as f
 
@@ -110,10 +110,12 @@ def test_unary_positive_for_daytime_interval():
     assert_unary_ast(data_descr, lambda df: df.selectExpr('+a'))
 
 @pytest.mark.parametrize('data_descr', ast_arithmetic_descrs, ids=idfn)
+@disable_ansi_mode
 def test_unary_minus(data_descr):
     assert_unary_ast(data_descr, lambda df: df.selectExpr('-a'))
 
 @pytest.mark.parametrize('data_descr', ast_arithmetic_descrs, ids=idfn)
+@disable_ansi_mode
 def test_abs(data_descr):
     assert_unary_ast(data_descr, lambda df: df.selectExpr('abs(a)'))
 
@@ -313,6 +315,7 @@ def test_bitwise_xor(data_descr):
             f.col('a').bitwiseXOR(f.col('b'))))
 
 @pytest.mark.parametrize('data_descr', ast_arithmetic_descrs, ids=idfn)
+@disable_ansi_mode
 def test_addition(data_descr):
     data_type = data_descr[0].data_type
     assert_binary_ast(data_descr,
@@ -322,6 +325,7 @@ def test_addition(data_descr):
             f.col('a') + f.col('b')))
 
 @pytest.mark.parametrize('data_descr', ast_arithmetic_descrs, ids=idfn)
+@disable_ansi_mode
 def test_subtraction(data_descr):
     data_type = data_descr[0].data_type
     assert_binary_ast(data_descr,
@@ -331,6 +335,7 @@ def test_subtraction(data_descr):
             f.col('a') - f.col('b')))
 
 @pytest.mark.parametrize('data_descr', ast_arithmetic_descrs, ids=idfn)
+@disable_ansi_mode
 def test_multiplication(data_descr):
     data_type = data_descr[0].data_type
     assert_binary_ast(data_descr,
@@ -373,6 +378,7 @@ def test_or(data_gen):
                        f.col('a') | f.col('b')))
 
 @ignore_order
+@disable_ansi_mode
 def test_multi_tier_ast():
     assert_gpu_ast(
         is_supported=True,

